### PR TITLE
Migrate documentation for all core functions from scalar_functions.md to code

### LINKED ISSUE #12801: Migrate documentation for all core functions from scalar_functions.md to code
### Is your feature request related to a problem or challenge?

As part of https://github.com/apache/datafusion/issues/12740 this ticket is to migrate all the existing documentation for core functions from the scalar_functions.md file to code and to remove from the scalar_functions.md file. In addition sql examples should be added for all of the updated functions.

This includes:

- arrow_cast
- arrow_typeof
- coalesce
- named_struct
- nullif
- nvl
- nvl2
- struct
- version

documentation should also be added for get_field which currently is not documented.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Omega359**
## Which issue does this PR close?

Closes #12801

## Rationale for this change

Documentation migration

## What changes are included in this PR?

docs, minor updates/improvements in getfield udf.

## Are these changes tested?

Yes.

## Are there any user-facing changes?

Docs updated.

### CODE REVIEW

File: docs/source/user-guide/sql/scalar_functions.md [hunk 1]
```diff
@@ -2806,93 +2751,10 @@ are not allowed
...
-```
-
-#### Arguments
-
-- **expression_n_name**: Name of the column field.
-  Must be a constant string.
-- **expression_n_input**: Expression to include in the output struct.
-  Can be a constant, column, or function, and any combination of arithmetic or
-  string operators.
+For more struct functions see the new documentation [`here`](https://datafusion.apache.org/user-guide/sql/scalar_functions_new.html)
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:2756]: this line seems to be repeated

```suggestion
```

**Review by alamb — COMMENTED** [review]
Thank you @Omega359  -- this looks great to me. I ran out of time to review this PR today but I will do tomorrow if no one beats me to it

**Comment by jonathanc-n** [comment]
@Omega359 This looks good to me!

File: datafusion/functions/src/core/getfield.rs [hunk 2]
```diff
@@ -133,7 +134,7 @@ impl ScalarUDFImpl for GetFieldFunc {
                     DataType::Struct(fields) if fields.len() == 2 => {
                         // Arrow's MapArray is essentially a ListArray of structs with two columns. They are
                         // often named "key", and "value", but we don't require any specific naming here;
-                        // instead, we assume that the second columnis the "value" column both here and in
+                        // instead, we assume that the second column is the "value" column both here and in
```
Comment (alamb) [inline datafusion/functions/src/core/getfield.rs:137]: ❤️

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @Omega359 and @jonathanc-n 

This PR has a merge conflict so I merged up from main to resolve it and plan to merge this PR when complete

**Pull request merged**